app/models/match.rb

```diff
@@ -4,4 +4,38 @@ class Match < ApplicationRecord
 
     validates :home_team, presence: true
     validates :away_team, presence: true
+
+    has_many :player_stats
+    has_many :players, through: :player_stats
+    has_many :stats, through: :player_stats
+
+    def all_goals
+        goal_id = Stat.where("stat_name = ?", "Goal").ids[0]
```

Comment:
better as 
```
goal_id = Stat.find_by(stat_name: 'Goal').id
```
`find_by` will always return the first matching item, and only one.